src/common/snippets/include/snippets/pass/mha_tokenization.hpp

```diff
@@ -0,0 +1,27 @@
+// Copyright (C) 2018-2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include <ngraph/pass/graph_rewrite.hpp>
+#include <ngraph/pattern/matcher.hpp>
+
+namespace ngraph {
+namespace snippets {
+namespace pass {
+
+/**
+ * @interface TokenizeMHASnippets
+ * @brief The pass tokenizes MHA-pattern into Subgraph
+ *        TODO: Write pattern
+ * @ingroup snippets
+ */
+class TokenizeMHASnippets: public ngraph::pass::MatcherPass {
+public:
+    TokenizeMHASnippets();
```

Comment:
General comment: As far as I understood, this pass could be logically divided into two parts:
1) Complex and flexible matching
2) Subgraph creation
So the question is: can we reuse subgraph creation from the general tokenization pipeline?
a) One way is to wrap `TokenizeSnippets` callback into a separate function and reuse it.
b) Another way is to transform `TokenizeMHASnippets` into a markup pass, and modify `TokenizeSnippets` matcher:
 `return (\*present conditions*\) || (GetSnippetsNodeType(n) == SnippetsNodeType::TokenizationForcedByPlugin);`

The motivation behind this question is that it could be not the last pattern-based tokenization we are enabling  (hopefully the last, but unlikely ;-). So I'm trying to come up with the most scalable solution.

@a-sidorova @dmitry-gorokhov what are your thoughts?